### ISSUE TITLE
[Feature Request] - Further customization of statuslights 

### ISSUE THREAD

**Issue by goomStar** [MilosKozak/AndroidAPS#2372]
As I like to personalize my statuslight threshold values for not only reservoir and battery level but also for the age of sensor, canula and battery, I usually adjust those in source (app/src/main/java/info/nightscout/androidaps/plugins/general/careportal/CareportalFragment.java lines 220 and following)

As levels are already accessible via plugin preferences, I would welcome to see the missing values in the same place. Certainly different people use different canulas, sensors and battery types, which they already have collected their personal experience with and surely would be happy to enhance the power of information the statuslights are showing them.

short form: I would love to see 'iage, cage, sage, bage' in the plugin preferences too

**Comment by swissalpine** [comment]
https://androidaps.readthedocs.io/en/latest/EN/Configuration/Preferences.html#overview

`Settings for status lights must be made in Nightscout settings. Set the following variables:

Cannula age: CAGE_WARN and CAGE_URGENT (standard 48 and 72 hours)
Insulin age (reservoir): IAGE_WARN and IAGE_URGENT (standard 72 and 96 hours)
Sensor age: SAGE_WARN and SAGE_URGENT (standard 164 and 166 hours)
Battery age: BAGE_WARN and BAGE_URGENT (standard 240 and 360 hours)
Treshold for warning reservoir level and critical reservoir level.

Treshold for warning battery level and critical battery level.`

**Comment by goomStar** [comment]
Exactly, I was wondering, why not all of those settings are accessible via the plugin.
https://androidaps.readthedocs.io/en/latest/EN/Configuration/Preferences.html#advanced-settings

Is there a specific reason why half of them are available "onboard/offline" and the others are "online only"?
I am trying to understand that - especially from a user perspective of someone, who tries to use NS as data-storage only and having all necessary settings directly within the app.

**Comment by swissalpine** [comment]
I think this is historically motivated. In the Careportal (today under Actions) IAGE, SAGE etc. were displayed and based on the Nightscout values they were colored, f. e. as a warning. Reservoir and battery levels are not found there, so there was no query of the values in AAPS realised. When extending the status lights with adding reservoir levels and battery levels  (instead of age) this query was not programmed (for convenience?), but settings were written instead.

**Comment by quizzmaster** [comment]
Nightscout doesn't have any settings for battery or reservoir levels which could be queried, so the local settings are needed in order to display these warnings.
